File: 2920.java
import java.util.Scanner;

public class Main{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int[] inputMelody = new int[8];
        int[] substractMelody = new int[7];
        int output = 0;
        
        for(int i = 0; i < 8; i++){
            inputMelody[i] = sc.nextInt();
        }
        for(int i = 0; i < 7; i++){
            substractMelody[i] = inputMelody[i+1]-inputMelody[i];
        }
        for(int i = 0; i < 6; i++){
            if(substractMelody[i] == substractMelody[i+1]){
                output = substractMelody[i];
            }
            else{
                output = 0;
                break;
            }
        }
        switch(output){
            case 0 :
                System.out.println("mixed");
                break;
            case 1 :
                System.out.println("ascending");
                break;
            case -1 :
                System.out.println("descending");
                break;
        }
    }
}

File: 2920.c
#include<stdio.h>

int main(){
    int inputMelody[8] = {0, };
    int melodyChecker[8] = {0, };
    int output = 0;
    for(int i = 0; i < 8; i++){
        scanf("%d", &inputMelody[i]);
    }
    for(int i = 0; i < 8; i++){
        melodyChecker[i] = inputMelody[i+1] - inputMelody[i];
    }
    for(int i = 1; i < 7; i++){
        if(melodyChecker[i] == melodyChecker[i-1]){
            output = melodyChecker[i];
        }
        else{
            output = 0;
            break;
        }
    }
    
    switch(output){
        case 0 :
            printf("mixed");
            break;
        case 1 :
            printf("ascending");
            break;
        case -1 :
            printf("descending");
            break;
    }
    return 0;
}
